src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -29,24 +24,29 @@ public class KeyVaultSecretsRepository : BaseSecretsRepository
         private const string FunctionKeyPrefix = "host--functionKey--";
         private const string SystemKeyPrefix = "host--systemKey--";
 
-        private readonly Lazy<KeyVaultClient> _keyVaultClient;
-        private readonly string _vaultName;
+        private readonly Lazy<SecretClient> _secretClient;
         private readonly IEnvironment _environment;
 
-        public KeyVaultSecretsRepository(string secretsSentinelFilePath, string vaultName, string connectionString, ILogger logger, IEnvironment environment) : base(secretsSentinelFilePath, logger, environment)
+        public KeyVaultSecretsRepository(string secretsSentinelFilePath, string vaultName, string clientId, ILogger logger, IEnvironment environment) : base(secretsSentinelFilePath, logger, environment)
         {
-            if (secretsSentinelFilePath == null)
+            if (string.IsNullOrEmpty(secretsSentinelFilePath))
             {
                 throw new ArgumentNullException(nameof(secretsSentinelFilePath));
             }
 
-            _vaultName = vaultName ?? throw new ArgumentNullException(nameof(vaultName));
+            if (string.IsNullOrEmpty(vaultName))
+            {
+                throw new ArgumentNullException(nameof(vaultName));
+            }
+
+            Uri vaultUri = new Uri($"https://{vaultName}{_environment.GetVaultSuffix()}");
 
-            _keyVaultClient = new Lazy<KeyVaultClient>(() =>
+            _secretClient = new Lazy<SecretClient>(() =>
             {
-                AzureServiceTokenProvider azureServiceTokenProvider = string.IsNullOrEmpty(connectionString) ? new AzureServiceTokenProvider()
-                    : new AzureServiceTokenProvider(connectionString);
-                return new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+                ManagedIdentityCredential credential = string.IsNullOrEmpty(clientId) ? new ManagedIdentityCredential()
```

Comment:
I worry about defaulting here to ManagedIdentityCredential, as that wouldn't work locally, technically. Managed Identity is purely a hosted platform feature. DefaultAzureCredential would allow for local development using the developer's identity in a consistent way.

Reply:
Great point, I'll use `DefaultAzureCredential` as the default

Reply:
Oh, should I stick to what I have then?